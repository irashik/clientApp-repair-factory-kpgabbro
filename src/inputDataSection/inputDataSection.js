// страница ввода данных о ремонте.

import React, { useEffect, useState } from "react";

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import DatePicker, { registerLocale, setDefaultLocale} from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";
import ru from 'date-fns/locale/ru';
registerLocale('ru', ru);
import * as log from 'loglevel';
log.setLevel('debug');

import InputRepairForm from "./inputRepairForm";
import ReadModuleList from '../readModuleList';



function InputDataSection(props) {

    const year =  new Date().getFullYear();
    const month = new Date().getMonth()
    const day = new Date().getDate();

    const [optedData, setOptedData] = useState(new Date(year, month, day));
    const [addedRepair, setAddedRepair] = useState([]);
    const [modalShow, setModalShow] = useState(false);

    
    // если нет в лок хранилище или срок использованиея..
    // проверка что вернуля не null? обработка ошибок?
    // await getUnitEquipmentList();
    //console.log('localstorage.setitem' + this.state.resultKeyListEquipment);
    //localStorage.setItem("resultKeyListEquipment", resultKeyListEquipment);
   



    function onSelectOptedData(selectedDate) {
        setOptedData(selectedDate);
        log.info(selectedDate);
    };

    function onHandleAddedRepair() {
        setAddedRepair([...addedRepair, 1]);
 
    };




    return (
        <Container fluid >
            <Row className="justify-content-md-center">
                <Col>
                    <h2 className="m-3">Что сегодня сделано?</h2>
                </Col>
                <Col>
                    <DatePickerDiv onSelectOptedData={onSelectOptedData} />
                </Col>
                <Col>
                    <Button variant="primary" 
                        id="AddRepairEquipment"
                        className="m-3" 
                        onClick={() => setModalShow(true)}
                        
                        >Добавить работы
                    </Button>
                </Col>
            </Row>

            <InputRepairForm    show={modalShow} 
                                onHide={() => setModalShow(false)}
                                handleAddedRepair={onHandleAddedRepair}
                                
                                />
            
            <ReadModuleList optedData={optedData} onAddedRepair={addedRepair}/>
        </Container>
    )
};
export default InputDataSection;


function DatePickerDiv(props) {

    // todo довать возможность выбора времени (интервалы полчаса)
    // доваить выбор начал и окончания ремонта

    //Locale with time

    const year =  new Date().getFullYear();
    const month = new Date().getMonth()
    const day = new Date().getDate();

    const [valueDate, setValueDate] = useState(new Date(year, month, day));
  

    function handlerOptedData(e) {
        setValueDate(e);    
        props.onSelectOptedData(e);
    };


    return (
            <DatePicker 
                className="m-3"
                locale="ru" 
                selected={valueDate}
                onChange={(date) => handlerOptedData(date)}
                dateFormat="dd MMMM yyyy"
                startDate={valueDate}
                id='StartDateValue' />
    );
};







        
      



