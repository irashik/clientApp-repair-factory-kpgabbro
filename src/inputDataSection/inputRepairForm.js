// форма (модальное окно) для ввода данных о ремонте.


import React, { useEffect, useState } from "react";
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Modal from 'react-bootstrap/Modal';
import CloseButton from 'react-bootstrap/CloseButton';
import Form from 'react-bootstrap/Form';
import { TiPen } from 'react-icons/ti';
import DatePicker, { registerLocale, setDefaultLocale} from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";
import { useDebouncedCallback } from 'use-debounce';
import ru from 'date-fns/locale/ru';
import {format, getMonth} from 'date-fns';

import Toast from 'react-bootstrap/Toast'


registerLocale('ru', ru);

import SearchList from "../searchListUnitEquipment";
import * as log from 'loglevel';
import { unloadInDb } from "../utils/loader";
log.setLevel('debug');


function InputRepairForm(props) {

    const [searchString, setSearchString] = useState('');
    const [filter, setFilter] = useState('');
    const [idEquipment, setIdEquipment] = useState('');

    const [repairCount, setRepairCount] = useState([]);
    const [materialCount, setMaterialCount] = useState([]);

    const [dataStart, setDataStart] = useState('');
    const [dataEnd, setDataEnd] = useState('');

    const [spendingJob, setSpendingJob] = useState('');

    let [repair, setRepair] = useState([]);
    let [material, setMaterial] = useState([]);
    let [value, setValue] = useState([]);



    const delay = 500;
    const debouncedSetFilter = useDebouncedCallback(
        filter => setFilter(filter),
        delay
    );

    const onChangeSearch = (e) => {
        const { value } = e.target;
        setSearchString(value);
        debouncedSetFilter(value);
        setIdEquipment(null)
    };
    
    const handlerSelectEquipment = (id, joinNameUnit, e) => {
        setSearchString(joinNameUnit);
        setIdEquipment(id);
    };

    function onClickAddedRepair() {


        // собрать данные из полей

        let materialJoin = material.map((i, a) => {
            let t = { name: i, value: value[a]};
            return t;
        });

        // подготовить данные для отправки на сервер
        const data = {
            dateRepairStart: dataStart,
            dateRepairEnd: dataEnd,
            equipment: idEquipment,
            repair: repair,
            author: localStorage.getItem('userId'),
            material: materialJoin,
            spendingJob: spendingJob
        };
        // записать в базу
        const url = process.env.HTTP_API_HOST + ":" + process.env.HTTP_API_PORT + "/equipment";

        unloadInDb(url, data)
            .then(result => {

                // все хорошо, сказать ок. и закрыть окно.
                // очистить поля остальные сами очищаются
                setSearchString('');
                setIdEquipment('');
                setSpendingJob('');

                // закрыть модальное окно.
                props.onHide();
                props.handleAddedRepair(); // для обновление списка ремонтов. 
                

                })
                .catch(err => {
                    alert('catch==' + JSON.stringify(err));
                    return new Error(err);
                });
    };
    

    function onHandleRepairCount() {
        setRepairCount([...repairCount, 1]);
    }
    function onHandleMaterialCount() {
        setMaterialCount([...materialCount, 1]);
    }
    const onSelectOptedDataStart = (selectedDate) => {
        setDataStart(selectedDate);
    }
    const onSelectOptedDataEnd = (selectedDate) => {
        setDataEnd(selectedDate);
    }
    function OnRecordRepair(record) {
        setRepair(record);
    }
    function OnRecordMaterial(record) {
        setMaterial(record);
    }
    function OnRecordValue(record) {
        setValue(record);
    };

      
    return   (    
        <Modal {...props} 
            backdrop="static"
            dialogClassName='modal-fullscreen'
            aria-labelledby="contained-modal-title-vcenter"
            animation={false}
        >
            <Modal.Header >
                <Modal.Title id="contained-modal-title-vcenter">
                    Форма ввода данных о работах
                </Modal.Title>
                <button type="button" className="btn-close" aria-label="Close"
                        onClick={props.onHide}

                ></button>
            </Modal.Header>
            <Modal.Body className="show-grid">
                <Container fluid id='input-module'>
                    <Row>
                        <Col>
                            <label>Время начало ремонта
                                <DatePickerDiv onSelectOptedData={onSelectOptedDataStart} />
                            </label>
                        </Col>
                        <Col>
                            <label>Время окончания ремонта
                                <DatePickerDiv onSelectOptedData={onSelectOptedDataEnd} />
                            </label>
                        </Col>
                    </Row>
                    <Row>
                        <Col >
                            <Form.Control 
                                    id='inputEquipment' size="sm" type="text" 
                                    placeholder="Выберите оборудование"
                                    value={searchString}
                                    onChange={onChangeSearch}
                                    idunit={idEquipment}
                                    />
                            <SearchList filter={filter} onSelectEquipment={handlerSelectEquipment} />
                        </Col>
                        <Col md={6}>
                            <FormInputRepair count={repairCount} onHandleRepair={OnRecordRepair}/>
                        </Col>
                        <Col md={2}>
                            <InputGroupButtonSmall name="equipment" onHandleRepairCount={onHandleRepairCount} />
                        </Col>
                    </Row>
                    <Row>
                        <Col >
                            <FormInputMaterial count={materialCount} onHandleMaterial={OnRecordMaterial} onHandleMaterialVal={OnRecordValue}/>
                        </Col>
                        <Col md={2}>
                            <InputGroupButtonSmall name="material" onHandleMaterialCount={onHandleMaterialCount} />
                        </Col>
                    </Row>
                    <Row>
                        <Form.Control id='inputSpendingJob' size="sm" type="text" 
                                        placeholder="Трудозатраты"
                                        value={spendingJob}
                                        onChange={(e) => setSpendingJob(e.target.value)}
                                        />
                    </Row>
                </Container>
            </Modal.Body>
            <Modal.Footer>
                <Button variant="primary" 
                            id="AddRepairEquipment"
                            className="m-3 d-grid gap-2" 
                            onClick={() => onClickAddedRepair()}
                            >
                    Записать
                </Button>
            </Modal.Footer>
        </Modal>        
    )
};
export default InputRepairForm;




function InputGroupButtonSmall(props) {

    let onAddedRecord = null;

    if(props.name === "equipment") {
        onAddedRecord = () => {
            props.onHandleRepairCount();
        }
    } 
    
    if (props.name === "material") {
       onAddedRecord = () => {
            props.onHandleMaterialCount();
        }
    }

    return (
        <Button variant="outline-dark" 
                data-toggle="tooltip" 
                className=""
                data-placement="top" 
                title="Добавить запись"
                id='btnAddDescription'
                onClick={() => onAddedRecord()}
                >+</Button>
    );
};

function DatePickerDiv(props) {

    // todo довать возможность выбора времени (интервалы полчаса)
    // доваить выбор начал и окончания ремонта

    //Locale with time

    const year =  new Date().getFullYear();
    const month = new Date().getMonth()
    const day = new Date().getDate();
    const hours = new Date().getHours();

    const [valueDate, setValueDate] = useState(new Date(year, month, day, hours));


  
    const handlerOptedData = (e) => {
        setValueDate(e);    
        props.onSelectOptedData(e);

    };

    useEffect(() => {
        props.onSelectOptedData(valueDate);
    }, []);

    if(props.type === 'dateStart') {

    }

    if(props.type === 'dateEnd') {

    }




    return (
        <React.Fragment>
            <DatePicker 
                            className="m-3"
                            locale="ru" 
                            selected={valueDate}
                            onChange={(date) => handlerOptedData(date)}
                            locale="ru"
                            showTimeSelect
                            timeFormat="p"
                            timeIntervals={30}
                            dateFormat="dd MMMM yyyy HH:mm"
                            startDate={valueDate}
                            id='StartDateValue' />
        </React.Fragment>
    );
};

function FormInputRepair(props) {
    const [count, setCount] = useState([]);
    const [repair, setRepair] = useState([]);


    function onHandleRepair(e, i) {
        let cloneRepair = [...repair];
        cloneRepair[i] = e;
        setRepair(cloneRepair);
        props.onHandleRepair(repair);
    }


    useEffect(() => {
        setCount([...count, 1])
        }, [props.count])


    function result(i) {
        return(
            <Form.Control id='inputRepairDescription' as='textarea' size="sm" rows={3} 
                placeholder="Что сделано?"
                key={i} 
                value={repair[i]}
                onChange={(e) => onHandleRepair(e.target.value, i)}
                />
        )
    }
    let arrayList = count.map((a,i) => result(i));
    return arrayList;
};

function FormInputMaterial(props) {

    const [count, setCount] = useState([]);
    const [material, setMaterial] = useState([]);
    const [value, setValue] = useState([]);



    function onHandleMaterial(e, i) {
        let cloneMaterial = [...material];
        cloneMaterial[i] = e
        setMaterial(cloneMaterial);
        props.onHandleMaterial(material);

   }

    function onHandleMaterialVal(e, i) {
        let cloneValue = [...value];
        cloneValue[i] = e;
        setValue(cloneValue);
        props.onHandleMaterialVal(value);
        
    }


    useEffect(() => {
        setCount([...count, 1]);
        
    }, [props.count])

    function result(i) {
        return (
            <Row key={'inputMaterial-'+i}>
                <Col>
                    <Form.Control id={'inputMaterialName-'+i} size="sm" type="text" 
                    placeholder="Введите материал"
                    key={i+'name'}
                    value={material[i]}
                    onChange={(e) => onHandleMaterial(e.target.value, i)}
                    />
                </Col>
                <Col md={3}>
                    <Form.Control id={'inputMaterialVal-'+i}  size="sm" type="text"  
                    placeholder="Введите его количество"
                    key={i+'val'}
                    value={value[i]}
                    onChange={(e) => onHandleMaterialVal(e.target.value, i)}
                    />
                </Col>
            </Row>
            )
        };

    let arrayList = count.map((a, i) => result(i));
    return arrayList;


};

