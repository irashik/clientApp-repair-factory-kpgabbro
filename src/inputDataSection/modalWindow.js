
import React, { useEffect, useState } from "react";

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Modal from 'react-bootstrap/Modal';
import ModalBody from 'react-bootstrap/ModalBody';
import ModalFooter from 'react-bootstrap/ModalFooter';
import ModalTitle from 'react-bootstrap/ModalTitle';
import ModalHeader from 'react-bootstrap/ModalHeader'


import Form from 'react-bootstrap/Form';
import { TiPen } from 'react-icons/ti';
import DatePicker, { registerLocale, setDefaultLocale} from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";
import { useDebouncedCallback } from 'use-debounce';

import ru from 'date-fns/locale/ru';
import {format, getMonth} from 'date-fns';

registerLocale('ru', ru);

//import SearchList from "./searchList";

import * as log from 'loglevel';
log.setLevel('debug');

import jQuery from 'jquery';

import InputModule from './inputRepairForm';




function InsertDataModal(props) {



    return (
      <Modal {...props} 
            
            size="xl"
            backdrop="static"
            aria-labelledby="contained-modal-title-vcenter">

        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Using Grid in Modal
          </Modal.Title>
        </Modal.Header>
        <Modal.Body className="show-grid">

        <InputModule props={props} />




          
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={props.onHide}>Close</Button>
        </Modal.Footer>
      </Modal>
    );
  }


  export default InsertDataModal;
