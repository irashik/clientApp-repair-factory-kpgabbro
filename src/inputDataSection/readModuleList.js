

import React, { useEffect, useState } from "react";

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Modal from 'react-bootstrap/Modal';
import ModalBody from 'react-bootstrap/ModalBody';
import ModalFooter from 'react-bootstrap/ModalFooter';
import ModalTitle from 'react-bootstrap/ModalTitle';
import ModalHeader from 'react-bootstrap/ModalHeader'


import Form from 'react-bootstrap/Form';
import { TiPen } from 'react-icons/ti';
import DatePicker, { registerLocale, setDefaultLocale} from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";
import { useDebouncedCallback } from 'use-debounce';

import ru from 'date-fns/locale/ru';
import {format, getMonth} from 'date-fns';

registerLocale('ru', ru);

//import SearchList from "./searchList";

import * as log from 'loglevel';
log.setLevel('debug');

import jQuery from 'jquery';




function ReadModuleList(props) {
    const [queryFromDb, setQueryFromDb] = useState([]);
    
    let url = new URL (process.env.HTTP_API_HOST + ":" + process.env.HTTP_API_PORT + "/equipment");
    url.searchParams.set("dateRepair", props.optedData);




    useEffect(() => {
        fetchListFromDb(url).then(queryFromDb => {
                setQueryFromDb(queryFromDb);
        });



    }, [props.repair, props.optedData]);


    /*
    для каждого элемента массива queryFromdb
    нужно отобразить элемент container c определенными полями.

    */

    log.info('array queryfromdb=' + queryFromDb.length);


    if(!queryFromDb.length) {
        return (
            <div>
                <h2>Нет записей за данное число</h2>
            </div>
        )
    } else {
        return (
            <React.Fragment>
                {   queryFromDb.map((i) => {
                    return <ReadModuleBlock i={i} /> 
                    })
                }
            </React.Fragment>
        )
    }
}

function ReadModuleBlock(props) {


    return (
        <Container fluid id='read-module' className="m-2">
        <Row>
            <Col sm={4} id='Readmoduleblock-equipment'>
                {props.i.equipment}

            </Col>
            <Col sm={4} id='readmoduleblock-repair'>
                {/* / add array function map */ }
                {props.i.repair}
            </Col>
            
            <Col sm={2}>
               <InputGroup.Checkbox aria-label="Checkbox for following text input" disabled/>

            </Col>
        </Row>


        { props.i.material.map((item) => {
                return (
                    <Row>
                        <Col sm={4} key={item._id}>            
                            {item.nameMaterial}
                        </Col>
                        <Col sm={4} >
                            {item.valueMaterial}
                        </Col>
                    </Row>
                );
        })}
     


     
               
        <Row>



            
            
            
            
            <Col sm={4} id='readmoduleblock-repairplan'>
                {props.i.repairPlan.map((item) => {
                    return(
                        <div key={item._id}>
                            {item.description}
                            {item.dateFinish}
                            {item.fihish}
                        </div>
                    )
                })}

            </Col>




            <Col sm={2}>
                <InputGroup.Checkbox aria-label="Checkbox for following text input" disabled/>
            </Col>
        </Row>

        <Row>
            <Col>
                <Button variant="outline-dark" 
                        data-toggle="tooltip" 
                        data-placement="top" 
                        title="Редактировать"
                        id='EditRepair'
                        item=''
                        >
                            <TiPen />
                        </Button>
            </Col>
        </Row>
        

    </Container>
    );
};

export default ReadModuleList;




async function fetchListFromDb(url) {

    const options = {
        method: 'GET',
        mode: 'cors',
        cache: 'no-cache',
        credentialls: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
        },
        redirect: 'follow',
      };

    const res = await fetch(url, options);
    return res.json();
};
