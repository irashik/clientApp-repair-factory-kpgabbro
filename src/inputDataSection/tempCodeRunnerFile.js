
function DatePickerDiv(props) {

    // todo довать возможность выбора времени (интервалы полчаса)
    // доваить выбор начал и окончания ремонта

    //Locale with time

    const year =  new Date().getFullYear();
    const month = new Date().getMonth()
    const day = new Date().getDate();

    const [valueDate, setValueDate] = useState(new Date(year, month, day));
  
    const handlerOptedData = (e) => {
        setValueDate(e);    
        props.onSelectOptedData(e);
    };

    useEffect(() => {
        props.onSelectOptedData(valueDate);
    }, []);


    return (
        <>
            <DatePicker 
                className="m-3"
                locale="ru" 
                selected={valueDate}
                onChange={(date) => handlerOptedData(date)}
                dateFormat="dd MMMM yyyy"
                startDate={valueDate}
                id='StartDateValue' />
        </>
    );