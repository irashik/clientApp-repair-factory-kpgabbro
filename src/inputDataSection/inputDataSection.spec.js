import React from 'react';
import { shallow, render } from 'enzyme';
import { InputDataSection } from './inputDataSection';




describe('<InputDataSection />', () => {

    test('snapShot correctly', () => {
        const component = shallow(<InputDataSection />);
        expect(component).toMatchSnapshot();

    });

});

