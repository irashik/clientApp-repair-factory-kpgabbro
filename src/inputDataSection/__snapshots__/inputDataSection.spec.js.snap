// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDataSection /> snapShot correctly 1`] = `
<Container
  fluid={true}
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <Col>
      <h2
        className="m-3"
      >
        Что сделано?
      </h2>
    </Col>
    <Col>
      <DatePickerDiv
        onSelectOptedData={[Function]}
      />
    </Col>
    <Col>
      <Button
        active={false}
        className="m-3"
        disabled={false}
        id="AddRepairEquipment"
        onClick={[Function]}
        variant="primary"
      >
        Добавить работы
      </Button>
    </Col>
  </Row>
  <InputRepairForm
    handleAddedRepair={[Function]}
    onHide={[Function]}
    show={false}
  />
  <ReadModuleList
    onAddedRepair={Array []}
    optedData={2021-11-18T21:00:00.000Z}
  />
</Container>
`;
